rest-service/manager_rest/security/user_handler.py

```diff
@@ -102,22 +90,45 @@ def get_token_status(token):
     :param token: The token to decrypt
     :return: A tuple: (expired, invalid, user, data)
     """
-    security = current_app.extensions['security']
-    serializer = security.remember_token_serializer
-    max_age = security.token_max_age
-
     user, data, error = None, None, None
-    expired, invalid = False, False
-
-    try:
-        data = serializer.loads(token, max_age=max_age)
-    except SignatureExpired:
-        expired = True
-    except (BadSignature, TypeError, ValueError) as e:
-        invalid = True
-        error = e
-
-    if data:
-        user = user_datastore.find_user(id=data[0])
+    expired, invalid = False, True
+
+    if token.startswith('ctok-'):
+        token_parts = token.split('-')
+        if len(token_parts) == 3:
+            _, tok_id, tok_secret = token_parts
+
+            sm = get_storage_manager()
+            token = sm.get(Token, tok_id, fail_silently=True)
+
+            error = 'Unauthorized.'
+            if token:
+                if verify_password(tok_secret, token.secret_hash):
+                    user = user_datastore.find_user(id=token._user_fk)
+                    # Legacy data structure
+                    data = [str(token._user_fk), 'no_hash_needed']
```

Comment:
this would be unnecessary, if we did the thing I suggested in one of the above comments, and pulled old-token `verfiy_token` into here. Then the result of this method wouldn't need `data` at all.

Actually, then we realize, instead of returning the `expired, invalid, error` from this, just for the caller to examine them and throw, this function could just throw. Then, `Authentication._authenticate_token` could be just removed.

Now my mouth is watering at the refactor possible here. Do ittttttt.